contracts/SwapNFT.sol

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.6;
+
+import {ERC721} from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import {ISwapNFT} from "./interfaces/ISwapNFT.sol";
+
+contract SwapNFT is ERC721, ISwapNFT {
+    address public immutable aelin;
+
+    uint256 public tokenCount;
+
+    mapping(address => bool) public hasMinted;
+
+    constructor(address _aelinAddress) ERC721("AelinSwapNFT", "SwapNFT") {
+        aelin = _aelinAddress;
```

Comment:
we can hardcode this value instead of passing it in

Reply:
it's fine to leave it like this in case we want to deploy on testnet